pkg/machines/libvirt-dbus.js

```diff
@@ -535,7 +536,20 @@ LIBVIRT_DBUS_PROVIDER = {
                         props.id = objPath;
                         props.connectionName = connectionName;
 
-                        dispatch(updateOrAddNetwork(Object.assign({}, props)));
+                        return call(connectionName, objPath, 'org.libvirt.Network', 'GetXMLDesc', [0], TIMEOUT)
+                                .then((xml) => {
```

Comment:
You can chain the promises. It looks better and also allows the catch to get the errors from all chained promises thought in your example catch will only get the outermost.